exporters/metrics/src/main/java/com/google/cloud/opentelemetry/metric/MetricDescriptorStrategy.java

```diff
@@ -0,0 +1,61 @@
+package com.google.cloud.opentelemetry.metric;
+
+import com.google.api.MetricDescriptor;
+
+import java.util.Collection;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.function.Consumer;
+
+/**
+ * The strategy for how to handle metric descriptors.
```

Comment:
Is the intention that users could implement this interface to handle metric descriptors in novel ways (such as filtering based on a prefix)? Or that they should use just the ones we provide? If it's the latter, an `enum` might be more appropriate here.

Reply:
I'm a fan of giving users flexibility when it's simple to do so.

i.e. this is my "escape hatch" to let folks fix issues in a release by overriding the metric descriptor synthesis/detection we have and do what they need.   I can lock it down to an enum, but we have time to decide on the right thing here, as this is still an "alpha" component.

Reply:
I brought up the enum question because the Readme suggests that this will eventually be configurable via environment variable. Is the idea that the env variable will let you pick one of the "existing" implementations, but you need to write custom code if you want to do something unusual? That makes sense.